### ISSUE TITLE
help with installation kali linux awus1900 RTL8814AU

### ISSUE THREAD

**Issue by gearcapitan** [aircrack-ng/rtl8812au#46]
I have tried to do every step for the installation of this driver, but apparently for more than try to make the RTL8814 = 1, it does not work, just install the RTL8812AU, help how do I install the correct driver ?, also the apt command -get install realtek-rtl88xxau-dkms, I also did not work, if the wifi icon appears, but when looking for wifi networks it shows absolutely nothing

https://ibb.co/gurxaw

**Comment by kimocoder** [comment]
make RTL8814=1 && make install RTL8814=1

I don't visit links btw, rather put images / outputs here.

**Comment by gearcapitan** [comment]
![captura de pantalla 103](https://user-images.githubusercontent.com/13740942/32397435-71f78440-c0af-11e7-9441-67fedec1d056.png)
![kali-2017-11-03-15-56-18](https://user-images.githubusercontent.com/13740942/32397475-99290f84-c0af-11e7-9b88-657f918e6681.png)
![kali-2017-11-03-15-56-37](https://user-images.githubusercontent.com/13740942/32397476-994c3572-c0af-11e7-9ad5-dba1931e4bb2.png)
![kali-2017-11-03-15-57-02](https://user-images.githubusercontent.com/13740942/32397492-aa98aad6-c0af-11e7-9558-27cfcabaf530.png)
![kali-2017-11-03-15-57-11](https://user-images.githubusercontent.com/13740942/32397494-aab4789c-c0af-11e7-83ca-91670ed560d7.png)

**Comment by kimocoder** [comment]
You dont need to use DKMS install when it's manually buildt.. it's installed as you can see.. what's the problem?

**Comment by gearcapitan** [comment]
the device does not appear, as if it were not connected

**Comment by gearcapitan** [comment]
![kali-2017-11-03-16-05-29](https://user-images.githubusercontent.com/13740942/32397717-d2a2b8cc-c0b0-11e7-8661-dcf05224e5d8.png)

**Comment by kimocoder** [comment]
well, I got the same adapter and it is working fine here. try unloading & loading through virtual machine again.

**Comment by gearcapitan** [comment]
![kali-2017-11-03-16-11-28](https://user-images.githubusercontent.com/13740942/32397902-a8c1c902-c0b1-11e7-81dc-522a37d9f0e8.png)
I already did what you told me, also try to restart but even so, it still does not appear

**Comment by gearcapitan** [comment]
I did the steps as you said, but in another virtual machine and now I run into this problem, it does not load the wifi networks 
![kali linux-2017-11-03-16-18-55](https://user-images.githubusercontent.com/13740942/32398117-cbd23cc8-c0b2-11e7-821c-074e22360442.png)

**Comment by kimocoder** [comment]
Have you enabled USB3 support in virtual machine then? Might want to use the extension pack for vmware

**Comment by gearcapitan** [comment]
if it is enabled
![captura de pantalla 104](https://user-images.githubusercontent.com/13740942/32398404-c276af40-c0b4-11e7-81ec-8356ddbe1d08.png)

**Comment by gearcapitan** [comment]
any idea?

**Comment by kimocoder** [comment]
Not really. Im using it in Kali (not VM) and it's working, so the problem could be VM .. leaving this issue open for others to discuss it. Anyway, output of dmesg after adapter plugin could be usefull too

**Comment by gearcapitan** [comment]
agree to then open another topic on this problem and hopefully it will be solved soon

**Comment by evilphish** [comment]
For other reasons I just downloaded a fresh copy of kali, created a virtual machine with Virtualbox,  cloned the github repo and built the 8814 module. The module loads up just fine and the dongle fires up. 
Also I am normally using the wifi dongle (8814 as well as 8812) in Linux Mint 18.3 and in Debian GNU/Linux 9. With both distributions the driver builds and loads just fine. 

@gearcapitan you are either doing something wrong while building / installing the module or something is not right with your kali install. In the case of the latter, you might try doing a fresh kali install to make sure you are on the same page.

**Comment by gearcapitan** [comment]
@evilphish Could you please kindly explain to me what you did? because the installation of virtualbox that appears in the other post, the installation I made of kali linux in virtualbox was the last iso in the page of kali, then I did an apt-upgrade and install everything before starting to install the driver
#47

**Comment by Mister-X-** [comment]
```apt-get upgrade``` will not upgrade the kernel, you need to do an ```apt-get dist-upgrade``` then reboot.

**Comment by evilphish** [comment]
@gearcapitan 
1. Downloaded http://cdimage.kali.org/kali-2017.2/kali-linux-2017.2-amd64.iso
2. Set up VM with USB3 controller, harddrive, etc
3. Install Kali
4. Upgrade Kernel to 4.13, either by installing (manually upgrading) linux-image-amd64 or as mister states do a dist-upgrade
5. install git
6. git clone the repo
7. make RTL8814=1
8. Either make install or for a quick test: cp 8814au.ko /lib/modules/<kernelversion> followed by depmod -a
9. connect my AWUS1900 to the usb port
10. right click the usb icon in the Virtualbox window and attach the wifi dongle
11. iwconfig wlan0 mode monitor
12. ifconfig wlan0 up
13. run wifite or whatever you like

**Comment by Mister-X-** [comment]
There is actually no need to install VM from ISO, there are prebuilt VM. So, just download the pre-built one and skip 1-3.

**Comment by gearcapitan** [comment]
![captura de pantalla 117](https://user-images.githubusercontent.com/13740942/32420561-b71d2c8e-c251-11e7-923d-73122c7115b9.png)
![captura de pantalla 118](https://user-images.githubusercontent.com/13740942/32420562-b737c94a-c251-11e7-828c-e51c05360388.png)
![captura de pantalla 119](https://user-images.githubusercontent.com/13740942/32420563-b74d8c8a-c251-11e7-81b9-7cfcef69c216.png)
![captura de pantalla 120](https://user-images.githubusercontent.com/13740942/32420564-b7635178-c251-11e7-94ec-e1d0a9b94aac.png)
????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????

**Comment by evilphish** [comment]
Did you unplug and replug your device after make install? Afaik installing the module does not necessarily load it immediately. Also more information would be appreciated. As in output of iwconfig, ifconfig, lsmod | grep 8814 and the last couple of lines of dmesg right after plugging in your wlan adapter.

**Comment by gearcapitan** [comment]
https://github.com/tpxSecurity/alfa_AWUS1900_adapter 

check this, is a tool actomatic installation driver, working for me

**Comment by kimocoder** [comment]
If your problem is to get it into monitor mode, then I could explain.. this driver does not support virtual interface (needed for airmon-ng), therefor it has to be det manually with

ifconfig wlan# (eg wlan1) down
iwconfig wlan# mode monitor
ifconfig wlan# up

That is what the mon0 script in the link does..

**Comment by nflanders** [comment]
I cannot for the life of me get my AWUS1900 to work on any build. 2017.2 following your steps @evilphish - nor using the automatic installation driver @gearcapitan 

Is this adaptor just not meant to work in kali!? Have spent the last 6 hours installing new kali virtual boxes, following all of the tutorials, etc. 

Will post my steps soon

**Comment by Mister-X-** [comment]
@nflanders Posting in the forum is more appropriate for that issue. This is a bug tracker, to track bugs

**Comment by nflanders** [comment]
@Mister-X- Got it. Thanks
